src/interpreter/terminator/intrinsics.rs

```diff
@@ -350,3 +335,54 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         self.tcx.normalize_associated_type(&f.ty(self.tcx, param_substs))
     }
 }
+
+fn numeric_intrinsic(name: &str, val: PrimVal) -> PrimVal {
```

Comment:
the only way to improve this is to use http://rust-num.github.io/num/num/trait.PrimInt.html, but we probably don't want any dependencies that would need to get pulled into rustc